components/layout/query.rs

```diff
@@ -766,10 +766,14 @@ pub fn process_resolved_style_request<'dom>(
     );
     let style = styles.primary();
     let longhand_id = match *property {
-        PropertyId::LonghandAlias(id, _) | PropertyId::Longhand(id) => id,
-        // Firefox returns blank strings for the computed value of shorthands,
-        // so this should be web-compatible.
-        PropertyId::ShorthandAlias(..) | PropertyId::Shorthand(_) => return String::new(),
+        PropertyId::NonCustom(id) => {
+            match id.unaliased().as_longhand() {
+                Some(id) => id,
+                // Firefox returns blank strings for the computed value of shorthands,
+                // so this should be web-compatible.
+                None => return String::new(),
+            }
```

Comment:
Could this be simply: `id.unaliased().as_longhand().unwrap_or_default()`?

Reply:
But there is no default LonghandId, the early return is needed.

Reply:
Ah, that makes sense!